prjxray/db.py

```diff
@@ -147,3 +161,16 @@ def get_tile_segbits(self, tile_type):
                 self.tile_types[tile_type.upper()])
 
         return self.tile_segbits[tile_type]
+
+    def get_required_fasm_features(self, part=None):
+        """
+        Assembles a set of required fasm features for given part. Returns a list
+        of fasm features.
+        """
+
+        # No required features in the db or part not given, return empty list
+        if self.required_features is None or part is None:
+            return set()
+
+        # Return list of part specific features
+        return self.required_features[part]
```

Comment:
Maybe make this `self.required_features.get(part, set())`, so if a part has not required features there isn't a problem?

Reply:
Fixed.